pennylane/devices/default_qutrit_mixed.py

```diff
@@ -85,10 +88,64 @@ def accepted_sample_measurement(m: qml.measurements.MeasurementProcess) -> bool:
     return isinstance(m, qml.measurements.SampleMeasurement)
 
 
+@qml.transform
+def warn_measurement_error_state(
+    tape: qml.tape.QuantumTape,
+) -> tuple[Sequence[qml.tape.QuantumTape], Callable]:
+    """If a measurement is an analytic state or density_matrix measurement and a readout error
+    parameter is defined wasns that the measurement error will not be applied.
+
+    Args:
+        tape (QuantumTape, .QNode, Callable): a quantum circuit.
+
+    Returns:
+        qnode (pennylane.QNode) or quantum function (callable) or tuple[List[.QuantumTape], function]:
+
+        The unaltered input circuit. The output type is explained in :func:`qml.transform <pennylane.transform>`.
+    """
+    if not tape.shots:
+        for m in tape.measurements:
+            if isinstance(m, qml.measurements.StateMP):
+                warnings.warn(f"Measurement {m} is not affected by measurement error.")
+
+    return (tape,), null_postprocessing
+
+
+def get_measurement_errors(damping_measurement_gammas, trit_flip_measurement_probs):
+    r"""Get the list of measurement errors that should be applied to each measured wire.
+
+    Args:
+        damping_measurement_gammas (List[float]): Gamma inputs for qml.QutritAmplitudeDamping channel
+            of the form :math:`[\gamma_{10}, \gamma_{20}, \gamma_{21}]`. This error models the expected
+            amplitude damping from longer measurements with shorter relaxation time associated with
+            transmon based qudits.
+        trit_flip_measurement_probs (List[float]): Gamma inputs for qml.QutritAmplitudeDamping channel
+            of the form :math:`[\p_{01}, \p_{02}, \p_{12}]`. This error models misclassification events
+            in measurement.
+
+    Returns:
+        measurement_errors (List[Callable]): List of measurement errors functions that should be
+        applied to each measured wire.
+    """
+    measure_funcs = []
+    if damping_measurement_gammas is not None:
+        if len(damping_measurement_gammas) != 3:
```

Comment:
You are correct, that validation is not necessary.

Reply:
The input validation, is not necessary, but I think it is helpful as without this the validation will occur after simulation. If it's a bigger simulation that may be pretty disappointing to find out after the simulation has finished. I added this validation to avoid this. Looking back I think it would be better to use the premade validation as there are other possible errors. I will add a new implementation.